lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -415,10 +415,12 @@ function _campaign_resource_reportback($nid, $values) {
     return $rb_info['rbid'];
   }
   else {
-    // Increment transaction id and log that the request has been received with new transaction id.
-    $incremented_transaction_id = dosomething_api_increment_transaction_id($headers['X-Request-Id']);
+    if ($headers['X-Request-Id']) {
```

Comment:
Originally, if it comes to the `else` portion, it would have meant that it was already sent to Rogue and this is when Rogue sends back to Phoenix. Rogue uses Gateway's API client which sets up the `X-Request-Id`. However, if it is coming from the API, we know that it doesn't have the `X-Request-Id` yet. So if the Rogue Collection flag is turned off but it is still coming from the API, the `X-Request-Id` header won't be set and the below lines will break.